setup.py

```diff
@@ -65,17 +100,76 @@ def get_extensions():
 
         include_dirs.append(ERFA_SRC)
 
+        # liberfa configuration
+        config_h = os.path.join(LIBERFADIR, 'config.h')
+        if not os.path.exists(config_h):
+            configure = os.path.join(LIBERFADIR, 'configure')
+            try:
+                if not os.path.exists(configure):
+                    subprocess.run(
+                        ['./bootstrap.sh'], check=True, cwd=LIBERFADIR)
+                subprocess.run(['./configure'], check=True, cwd=LIBERFADIR)
+            except (subprocess.SubprocessError, OSError) as exc:
+                warn(f'unable to configure liberfa: {exc}')
+
+            if (not os.path.exists(configure) and 'sdist' in sys.argv and
+                    sys.platform != 'win32'):
+                raise RuntimeError(
+                    'missing "configure" script in "liberfa/erfa"')
+
+        if os.path.exists(config_h):
+            include_dirs.append(LIBERFADIR)
+            define_macros.append(('HAVE_CONFIG_H', '1'))
+        else:
+            define_macros.extend(get_liberfa_versions())
+
     erfa_ext = NumpyExtension(
         name="erfa.ufunc",
         sources=sources,
         include_dirs=include_dirs,
         libraries=libraries,
+        define_macros=define_macros,
         language="c")
 
     return [erfa_ext]
 
 
-VERSION_TEMPLATE = """
+def _guess_next_dev(version, liberfadir=None):
```

Comment:
Over at astropy, we've changed the usage of the `setuptools_scm` quite a bit, making it simpler. What do you think? See https://github.com/astropy/astropy/pull/10774

Reply:
Yes, I like the solution adopted in astropy. I think it is better if I implement it in a different PR, do you agree?
In any case the machinery to guess the next dev version should be propagated in all places where ``setuptools_scm`` is used :/.

Reply:
Yes, makes sense to do in a separate PR.